File: LeetCode/C++/Easy/Split a String in Balanced Strings/onePass.cpp
class Solution {
public:
	int balancedStringSplit(string s) {

		int r = 0, ans = 0;

		for (int i = 0; i < s.length(); i++) {
			s[i] == 'R' ? r++ : r--;
			if (r == 0) ans++;
		}

		return ans;
	}
};

File: LeetCode/Python3/Easy/Split a String in Balanced Strings/onePass.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        
        r = 0
        ans = 0
        
        for c in s:
            if c == 'R':
                r += 1
            else:
                r -= 1
                
            if r == 0:
               ans += 1
            
        return ans